homeassistant/components/nuki/lock.py

```diff
@@ -1,54 +1,66 @@
-"""Nuki.io lock platform."""
+"""
+Nuki.io lock platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/lock.nuki/
+"""
 from datetime import timedelta
 import logging
+import requests
 
 import voluptuous as vol
 
-from homeassistant.components.lock import DOMAIN, PLATFORM_SCHEMA, LockDevice
+from homeassistant.components.lock import (
+    DOMAIN, PLATFORM_SCHEMA, LockDevice, SUPPORT_OPEN)
 from homeassistant.const import (
-    ATTR_ENTITY_ID, CONF_HOST, CONF_PORT, CONF_TOKEN)
+    ATTR_ENTITY_ID, CONF_HOST, CONF_PORT, CONF_TOKEN, CONF_TIMEOUT)
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.service import extract_entity_ids
 
-REQUIREMENTS = ['pynuki==1.3.2']
-
 _LOGGER = logging.getLogger(__name__)
 
 DEFAULT_PORT = 8080
+DEFAULT_TIMEOUT = 5
```

Comment:
Why does the timeout needs to be configurable?

Reply:
With the Nuki lock you can use an Android phone as bridge. That's my case. If I set 5s the timeout, most of the messages between HA and the bridge get lost. I'm not sure if it's due to the bridge app, the wifi communication with the mobile, my android device, ... Whatever it is, with a higher timeout value these problems disappear.

Reply:
Right, but then why shouldn't we just increase the timeout? Now you're making it configurable so you can fix your problems, instead of fixing it for all the users out there

Reply:
There is a dedicated hardware bridge from Nuki, which I don't own. I guess this device should be faster and more reliable than the software bridge running on the phone. So, probably, in these cases, a default timeout of 5s should be fine. So my proposal is: if your connection between HA and the bridge is ok, don't do anything (5s timeout default). If not, try to increase this value.

Reply:
I have the hardware bridge and 5s also leads to a timeout although it should work. In the current state, using the nuki component in hassio is useless for me. When trying to update the lock state, the timeout error blocks any communicatione/state update for the next 10 minutes. I second franfos idea to make the timeout value configurable.

Reply:
Would 10 seconds timeout work better? What timeout are you using?

Reply:
I'm currently using 15s. Works most of the times, but it still gets some timeouts.

Reply:
Ok. I think a timeout option seems necessary here. @balloob?

Reply:
Why can't we just set the timeout 20s?

Reply:
In the case that the communication works fine in someone's setup (i. e. 5s timeout it's ok), if you have any puntual problem in that communication, it will take 20s every time to detect it. I think in that case you are punishing that user for someone else problem. 
In my case, probably a fixed 20s timeout would be a good solution, but maybe it's not good for everyone. I think it can fit in more cases, as I also have seen a variable timeout configuration in other components.

Reply:
> have seen a variable timeout configuration in other components.

Yes and we're phasing it out. The problem is that we're putting a problem on our users because we're afraid to make decisions as developers. Now each user has to fiddle with this.

Let's just put it at 20s and see if someone complains ?

Reply:
Ok. So, do I set a default 20s timeout, but with the option to change it or not? I mean, do I remove completely the timeout configuration parameter?

Reply:
Please remove the timeout config option.

Reply:
Done